src/config.ts

```diff
@@ -148,7 +151,10 @@ class ConfigurationFactory {
         const url = githubEvent["repository"]
           ? githubEvent["repository"]["url"]
           : `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}`
-        return `${url}/tarball/${ref}`
+        const urlTarball = `/tarball/${ref}`
+        const encodedTarball = encodeURIComponent(urlTarball)
```

Comment:
I would exclude `/tarball/` from the encoded part. I think that aligns with the point @aoterolorenzo has already brought up in the [discussion](https://github.com/vmware-labs/vmware-image-builder-action/pull/159#discussion_r1328557062).

Additionally, it might be worth considering if it makes sense to exclude `/` and `-` characters from that encoding. In this [PR](https://github.com/bitnami/charts/pull/18754) branch example, `https://github.com/bitnami/charts/tree/feat/%2312580-add-cronjob-for-matomo-chart`, does not encode them.